src/view/widgets/reset_button.py

```diff
@@ -0,0 +1,132 @@
+        each widget to its specified reset value. Additionally, resets
+        certain identifiers and page number in the controller and updates
+        the item grid and floating elements in the reco explorer app if needed.
+
+        :param event: The event that triggered the reset
+        :param block: The block containing the widgets to be reset
+        :return: None
+        """
+        for widget in block:
+            if widget.value is not widget.params.get("reset_to"):
+                widget.value = widget.params.get("reset_to")
+
+        self.controller_instance.reset_defaults(event.obj.params["resets"])
+        self.controller_instance.reset_page_number()
+
+        reset_identifiers_item = [
+            c.RESET_IDENTIFIER_MODEL_CHOICE,
+            c.RESET_IDENTIFIER_ITEM_CHOICE,
+            c.RESET_IDENTIFIER_UPPER_ITEM_FILTER,
+            c.RESET_IDENTIFIER_ITEM_FILTER
+        ]
+
+        reset_identifiers_reco = [
+            c.RESET_IDENTIFIER_RECO_FILTER,
+            c.RESET_IDENTIFIER_UPPER_RECO_FILTER
+        ]
+
+        for reset_type in event.obj.params["resets"]:
+            if reset_type in reset_identifiers_item:
+                self.reco_explorer_app_instance.item_grid.objects = {}
+                self.reco_explorer_app_instance.floating_elements.objects = []
+                self.reco_explorer_app_instance.draw_pagination()
+            elif reset_type in reset_identifiers_reco:
+                self.reco_explorer_app_instance.get_items_with_parameters()
+
+    def get_widgets_to_reset(self, block):
+        """
+        Get the list of widgets to reset for a given block.
+
+        :param block: The block from which to collect the widgets.
+        :return: The list of widgets to reset.
+        """
+        widgets_to_reset = []
+        for widget in block:
+            widgets_to_reset.extend(self.collect_leaf_widgets(widget))
+        return widgets_to_reset
+
+    def collect_leaf_widgets(self, widget):
+        """
+        This method recursively traverses the widget hierarchy starting from
+        the given widget, and collects all the leaf widgets. A leaf widget is
+        a widget that does not have any children.
+
+        If the given widget is an instance of any of the layout widget types
+        defined in `self.layout_widget_types`, this method will traverse its
+        children and collect the leaf widgets from them
+        * recursively.
+
+        If the given widget is an instance of any of the leaf widget types
+        defined in `self.leaf_widget_types`, it will be added to the list
+        of leaf widgets, unless it has a `reset_identifier
+        *` attribute.
+
+        The resulting list of leaf widgets is returned by this method.
+
+        :param widget: The widget to collect the leaf widgets from.
+        :return: A list of leaf widgets.
+        """
+        leaf_widgets = []
+        if isinstance(widget, self.layout_widget_types):
```

Comment:
Ich denke es wäre eine Überlegung wert "is_leaf" als property auf dem widget selbst zu haben dann muss man das hier nicht nochmal in ner Liste Pflegen sondern nur einmal auf dem widget.

Reply:
Eine gute Idee, hab ich jetzt mal so gemacht.